### DISCUSSION TITLE
get host ip

### DISCUSSION THREAD

**Discussion by githubcheng2978** [rabbitmq/rabbitmq-server#3252] [Other]
how dose rabbitmq server obtain get host ip of the client when it is deployed in the container

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This is a bit light on context. For every connection, RabbitMQ has an open socket and can obtain peer information (including hostname and client connection port) from it. If the network connection is from is a [Proxy protocol-aware](https://www.rabbitmq.com/networking.html#proxy-protocol) too (e.g. an HAproxy),
the provided Proxy protocol header is used instead.
